src/BlockHorizons/BlockPets/listeners/EventListener.php

```diff
@@ -97,10 +97,12 @@ function(array $petData) use($player, $loader): void {
 							(bool) $data["Visible"],
 							$data["Inventory"]
 						);
-						$pet->spawnToAll();
-						$pet->setDormant(false);
-						if($hard_reset) {
-							$pet->close();
+						if(!is_null($pet)){
```

Comment:
if($pet !== null)

Reply:
How about issue when pet dies? a lot of people have a problem with that

Reply:
I have no idea i haven't had the time to look at it. I fixed the majority of bugs so the plugin was usable

Reply:
https://github.com/BlockHorizons/BlockPets/pull/231/commits/f9672a12db7d4e1f7e06ec9a8d1c622f1c7a8b71